src/translator/html.cpp

```diff
@@ -463,24 +295,41 @@ HTML::HTML(std::string &&source, bool process_markup) {
         break;
 
       case markup::Scanner::TT_TEXT: {
+        // If the previous segment was the open or close tag of a block element
+        // we treat the text after it as a new sentence.
+        if (addSentenceBreak) {
+          if (!(source.empty() || (source.size() > 2 && source.substr(source.size() - 2) == ""))) {
+            stack.push_back(makeTag({Tag::WHITESPACE}));
+            // Important: span->size() == 0 to make it behave as a void element.
+            // Also important: position before the \n\n tokens, not after, to
+            // make it easier to remove them later through apply().
+            spans_.push_back(Span{source.size(), source.size(), stack});
+            source.append("\n\n");  // TODO assumes ssplit-mode = wrapped_text
```

Comment:
split-mode default isn't `wrapped_text`, it's `paragraph`. But I don't think this affects the ultimate outcome when HTML.

Reply:
Hmm, you're right. I've misjudged where that default came from. `\n\n` will still be picked up correctly, but now I'm a bit worried about the quality of [pages that wrap their HTML manually](https://neural.mt) because those `<p>...</p>` will now be interpreted as multiple paragraphs?